### PR TITLE
Fix Omega-Reset-Mu bug

### PR DESCRIPTION
**Pull request by ashleybaldock**
Added a further change so the clearDeck/Hand functions clear the hand/deck in the same way lua does- replacing them with a new object. 

I'll try and make a test case to cover this scenario.

### CODE REVIEW

**Comment by salinecitrine** [comment]
Testing this on the preview deployment, it shows "Iteration limit reached". Initially I thought that this meant it wasn't working, but maybe it's correct? That result would indicate that the wand never reloads (recharges). Do you know if that's also the case in-game? Would be interesting if it let you ignore recharge time on a wand.

**Comment by ashleybaldock** [comment]
Isn't that one of Wand Refresh's quirks - if you only call it once in a cast it lets you skip all recharge delay? I'll check if it does that in-game.

This could be a useful warning/info message actually - the `Iteration limit reached` message hints at it, but explicitly showing a message that the wand isn't recharging would be good.

**Comment by ashleybaldock** [comment]
(Oh also from what I understand, it lets you ignore recharge delay, but the delay is still stored in the wand. So if you have a deficit built up over time you can end up with one huge recharge delay later, e.g. if you accidentally fail to cast Wand Refresh one time (or cast it twice). The Recharge Delay decreasing by .42 each cast seems to fit with that.)

**Comment by salinecitrine** [comment]
You're right, that makes more sense. I was getting confused because I thought that Wand Refresh wasn't doing anything, but it is actually called. The sim might have a bug [here](https://github.com/salinecitrine/noita-wand-simulator/blob/5d9d27443f0e4238e0acdf1f7dcc33ac2ca6ce85/src/app/calc/eval/clickWand.ts#L245), where it checks if it should end simulation after a Wand Refresh. It only checks the last action called, but in this unusual case, Wand Refresh gets called, another action is called, and /then/ the cast ends. So this check fails.

**Comment by salinecitrine** [comment]
```
    if (
      !fireUntilReload ||
      (endOnRefresh &&
        calledActions!.filter((a) => a.action.id === 'RESET').length > 0) ||
      calledActions!.length === 0
    ) {
      break;
    }
```

seems to fix the iteration limit issue

**Review by salinecitrine — APPROVED** [review]

**Comment by ashleybaldock** [comment]
Another case I can think of where you can have a spell called after Wand Refresh is [Tau, Wand Refresh, _Something_](https://noita-wand-simulator.salinecitrine.com/?actions_per_round=1&deck_capacity=26&reload_time=0&shuffle_deck_when_empty=false&cast_delay=0&mana_charge_speed=20000&mana_max=20000&spread=0&spells=,,,,,,,,,TAU,RESET,LIGHT_BULLET)

File: src/app/calc/__generated__/gun_actions.ts [hunk 1]
```diff
@@ -7430,7 +7427,7 @@ export const actions: Action[] = [
 						
 						if (( check != null ) && ( ( check.type === ACTION_TYPE_PROJECTILE ) || ( check.type === ACTION_TYPE_STATIC_PROJECTILE ) || ( check.type === ACTION_TYPE_MATERIAL ) || ( check.type === ACTION_TYPE_UTILITY ) ))  {
 							valid = true
-							break
+							return false;
```
Comment (ashleybaldock) [inline src/app/calc/__generated__/gun_actions.ts:7430]: This seems to be what broke Add Trigger - didn't spot that before :)

**Pull request merged**